### ISSUE TITLE
Automate sending email to users to let them know their current usage

### ISSUE THREAD

**Issue by joachimweyl** [CCI-MOC/process_csv_report#8]
## Motivation
Regularly let users know how many resources they have used prior to receiving their official invoice.

## Completion Criteria
email sent out weekly with current invoice usage data.

## Description
- [ ] Create a system that scrapes the PI emails from the invoice csv
- [ ] Take the Invoice grouped by PI data from the previous Issue and add it to the email as an attached CSV or inline, whichever is easier.
- [ ] Create an automation that will send the email at a specific time (once a month mid month).
- [ ] Test with @joachimweyl or @Milstein
- [ ] create PR to automate

## Completion dates
Desired - 2024-02-14
Required - TBD

**Comment by joachimweyl** [comment]
@msdisme do we need some way to remove PIs from the list if they do not wish to receive this email? Or shall we just tell people they should filter it to trash if they don't want it?

**Comment by msdisme** [comment]
Let's get a third opinion from wayne

**Comment by joachimweyl** [comment]
@waygil what are your thoughts on the need to allow PIs to remove themselves from an automated weekly email that provides current usage?

**Comment by joachimweyl** [comment]
Any cadence can be set. The desire for these emails was to alleviate the pain of PIs not knowing how big their bill would be beforehand. Will they provide the desired effect every half month?
That also raises a good point, we don't need to send these the week the bill is sent out as the bill will provide this data. so maybe something like once a month in the middle of the month, or twice a month 1/3 and 2/3 of the way through.

**Comment by waygil** [comment]
Good point Kim.  I would then think a mid  month email might suffice as they will get an invoice at the beginning of the next month.

**Comment by joachimweyl** [comment]
Updated description to reflect this.

**Comment by msdisme** [comment]
My concern is that GPUs increase at a rapid rate per day.  Perhaps it should be for each time it goes up by $100?

**Comment by joachimweyl** [comment]
@msdisme any dollar amount instead of cron job automation is going to greatly change the way this automation works.

**Comment by msdisme** [comment]
The problem is that a PI's student can forget to sign off one GPU  and run up  ~$50 a day, 4 gpu's=$200 a day and the PI won't know it.   So we need a way to notify that does not create so much noise that people want to cancel their notifications.     

@knikolla am I recalling correctly that you thought one option might be to put a version of a csv with a running balance (daily?) and send regular mail saying "look here to see your current balances for the month."  Would that be a lighter lift?

**Comment by joachimweyl** [comment]
@QuanMPhm a few thoughts have been thrown around how best to communicate large increases in usage to PIs. It sounds like we might need to think up a way to track usage so we know when there is a large change. I know you worked on a way to break the CSV into PI-specific CSVs. Is there an easy way to check those CSVs to find when a large change happens and send emails based on a large change?

@msdisme if this is possible what large change to we want to look out for? increases of $100 in a day?

**Comment by joachimweyl** [comment]
I am confirming with @msdisme. @waygil or @syockel please also feel free to chime in what you think would be a large change that we need to alert users to. Is it a % change or a dollar amount? How often do we want to check, every day?

Here is an example of what it could be like
1. Starting 2X days after the beginning of the month check the previous X days
1. If Y(% or dollar amount) changes since the last X days send an email to the user

if we assume 1 day and 50%
1. Staring at the end of day 2 compare it to 1
1. If there is a 50% change since day one send the user an email

if we assume 2 days and $500
1. Staring at the end of day 4 compare it to 2
1. If there is a $500 change since days one & two send the user an email

**Comment by msdisme** [comment]
We ended up on an email solution because we do not currently have a "go look here" solution.   It's been enough time that we should confirm LOE (Level Of Effort) vs. getting a to a "go look here" approach.
